Sources/NIO/Bootstrap.swift

```diff
@@ -483,29 +483,56 @@ public final class DatagramBootstrap {
 
     private func bind0(eventLoopGroup: EventLoopGroup, to address: SocketAddress) -> EventLoopFuture<Channel> {
         let eventLoop = eventLoopGroup.next()
-        let channelInitializer = self.channelInitializer ?? { _ in eventLoop.newSucceededFuture(result: ()) }
-        let channelOptions = self.channelOptions
 
         let promise: EventLoopPromise<Channel> = eventLoop.newPromise()
         do {
             let channel = try DatagramChannel(eventLoop: eventLoop as! SelectableEventLoop,
-                                                    protocolFamily: address.protocolFamily)
+                                              protocolFamily: address.protocolFamily, activeWithoutBind: false)
+            initDatagramChannel(channel: channel, address: address, promise: promise)
+        } catch let err {
+            promise.fail(error: err)
+        }
 
-            channelInitializer(channel).then {
-                channelOptions.applyAll(channel: channel)
-            }.then {
-                channel.register()
-            }.then {
-                channel.bind(to: address)
-            }.map {
-                channel
-            }.cascade(promise: promise)
+        return promise.futureResult
+    }
+
+    /// Open a `DatagramChannel` without binding it.
+    ///
+    /// This `DatagramChannel` can only be used to send data until `bind` is explicitly called.
```

Comment:
This is not right.

If you send on a `SOCK_DGRAM` socket without explicitly binding it, it does what`SOCK_STREAM` does when you `connect` without previously calling `bind`: the OS selects and binds an address for you. This means that you *can* receive data immediately after you start sending it.

If we want to support this use-case, that's fine, but I think we should reword this documentation and also add an explicit test to check that the `DatagramChannel` does behave correctly in this case.

Reply:
@Lukasa actually I did not know that it "binds" under the hood for you. Knowing this maybe we should just not care about this and close the PR without merging it ? @weissi WDYT ?